packages/swingset-liveslots/src/virtualObjectManager.js

```diff
@@ -824,6 +824,10 @@ export function makeVirtualObjectManager(
       // kdebug(`vo make ${baseRef}`);
 
       const initialData = init ? init(...args) : {};
+      if (typeof initialData !== 'object') {
+        Fail`initial data must be object, not ${initialData}`;
+        // a common mistake is to use `() => {foo:1}`, not `() => ({foo:1})`
+      }
```

Comment:
I'm not sure this does quite what you want.  I tried fiddling around with this in node, and I think the error message you'd get would look like `initial data must be object, not () => { foo: 1 }`, to which the programmer may likely think, "yes, that's what I said".

Reply:
Hm, maybe there are three cases, not just the two I was thinking of.

The good case, where each new VO instance is created with the same `{foo: 1}` state, ignoring any arguments:

```js
const initialize = () => ({foo: 1});
const makeThing = defineKind('name', initialize, behavior);
```

The first bad case, where the missing parentheses creates non-object initial data (`{foo: 1}` in that context actually evaluates to `undefined`):
```js
const initialize = () => {foo: 1};
const makeThing = defineKind('name', initialize, behavior);
```

Without the new `typeof` check, the error message happens when `Object.getOwnPropertyNames(initialData)` is confronted with `undefined`. I get `Cannot convert undefined or null to object`. *With* the check, I get `initial data must be object, not "[undefined]"`. Which, admittedly, is not a huge improvement, but at least it mentions "initial data", which should be a clue.

I'm guessing the error you tried out was:
```js
const initialize = () => {foo: 1};
const makeThing = defineKind('name', () => initialize, behavior);
```

which, for me (in an AVA unit test), gets an error like `initial data must be object, not "[Function initialize]"`, but if the error-displaying pathway had decided to stringify the function, then I imagine I'd have gotten something like what you described, which *would* be confusing. It'd say "initial data" in the error, but it's not immediately obvious that it's receiving a function, rather than the return value of your function.

I'm gonna claim that any improvement here is an improvement :)